lib/dns.js

```diff
@@ -240,9 +243,17 @@ function onresolve(err, result, ttls) {
 }
 
 
-function resolver(bindingName) {
-  var binding = cares[bindingName];
+class Channel {
+  constructor() {
```

Comment:
Maybe not in this PR. @XadillaX already mentioned above that there might be other options we could consider adding here, so it seems something to give a little more thought to :)